### PR TITLE
refactor: port over some changes from Deep Module experiment

### PR DESCRIPTION
**Pull request by wKovacs64**
This pull request introduces a significant refactor and improvement to how drink mutations (create, update, delete) are handled in the codebase, centralizing logic in a new `mutations.server.ts` module, improving test coverage, and simplifying the admin drink management routes. It also adds new environment variables for Fastly cache purging and updates the search loader for efficiency. The most important changes are grouped below:

### Drink mutation logic refactor and test coverage

* Introduced a new module, `app/drinks/mutations.server.ts`, which centralizes the logic for creating, updating, and deleting drinks, including image handling and cache purging. This module ensures consistent error handling and cache invalidation across all mutations.
* Added comprehensive tests for drink queries, mutations, and validation in `app/drinks/mutations.test.ts`, covering scenarios such as image upload failures, slug changes, and validation edge cases.
* Refactored admin drink edit and delete routes to use the new mutation functions, removing duplicated logic and improving error feedback to users. (`app/routes/admin.drinks.$slug.edit.ts`, `app/routes/admin.drinks.$slug.delete.ts`) [[1]](diffhunk://#diff-395e9ed08b85f35fbe8d2ddf94fa840617a9c36e736740c23f7686fb65628f60L3-R7) [[2]](diffhunk://#diff-b7ce649b4de84f01a7ceffd363ee3b43f82abda4e085497e4ab425dd82bfe2a3L3-R4) [[3]](diffhunk://#diff-b7ce649b4de84f01a7ceffd363ee3b43f82abda4e085497e4ab425dd82bfe2a3L18-L36)

### Model and utility changes

* Renamed model-layer functions in `app/models/drink.server.ts` for clarity: `createDrink` → `insertDrinkRow`, `updateDrink` → `updateDrinkRow`, and `deleteDrink` → `deleteDrinkRow`. This clarifies their role as low-level DB operations, distinct from the new mutation logic. [[1]](diffhunk://#diff-c179dcfb7220c80d8d8d7557c1f00e7141a0bd1b1f50eb06339140db20df1a78L56-R56) [[2]](diffhunk://#diff-c179dcfb7220c80d8d8d7557c1f00e7141a0bd1b1f50eb06339140db20df1a78L71-R71) [[3]](diffhunk://#diff-c179dcfb7220c80d8d8d7557c1f00e7141a0bd1b1f50eb06339140db20df1a78L88-R88)
* Added unit tests for authentication utility functions in `app/auth/utils.test.ts`, improving coverage for redirect safety and return-to URL creation.

### Search and environment improvements

* Updated the search loader (`app/routes/_app.search/route.tsx`) to use the new `searchDrinks` signature, simplifying the data flow and removing unnecessary intermediate steps. [[1]](diffhunk://#diff-1cb1a3a8aea35e88357f392916bc571cd05003bb0e2b1648ac1680922647a20eL7-L10) [[2]](diffhunk://#diff-1cb1a3a8aea35e88357f392916bc571cd05003bb0e2b1648ac1680922647a20eL43-R41) [[3]](diffhunk://#diff-1cb1a3a8aea35e88357f392916bc571cd05003bb0e2b1648ac1680922647a20eL57-L67)
* Added Fastly cache purge environment variables to `.env.test` for improved test configuration and cache management.

### Admin form simplification

* Removed the hidden input for `existingImageUrl` from the `DrinkForm` component, since the new mutation logic manages image state.

### CODE REVIEW

File: app/search/minisearch.server.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import MiniSearch from "minisearch";
 import type { Drink } from "#/app/db/schema";
+import { getSearchData } from "./cache.server";
```
[the remark this answers is not in the document]
Comment (wKovacs64) [reply app/search/minisearch.server.ts:4]: Both imports are used inside function bodies, not at module evaluation time — ESM resolves this fine. A third module would add indirection with no practical benefit.

File: app/search/minisearch.server.ts [hunk 2]
```diff
@@ -33,7 +34,11 @@ export function createSearchIndex(drinks: Drink[]): MiniSearch<SearchableDrink>
...
 }
 
-export function searchDrinks(searchIndex: MiniSearch<SearchableDrink>, query: string): string[] {
+export async function searchDrinks(query: string): Promise<Drink[]> {
+  const { allDrinks, searchIndex } = await getSearchData();
   const results = searchIndex.search(query, { combineWith: "AND" });
-  return results.map((result) => result.slug);
+
+  return results
+    .map((result) => allDrinks.find((drink) => drink.slug === result.slug))
```
[the remark this answers is not in the document]
Comment (wKovacs64) [reply app/search/minisearch.server.ts:42]: Dataset is <50 items and unlikely to ever reach 100 (per project scope). The linear scan is clearer and plenty fast — a Map would be premature optimization here.

File: app/drinks/mutations.server.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  });
+
+  try {
+    purgeSearchCache();
+    await purgeDrinkCache({ slug: drink.slug, tags: drink.tags });
+  } catch (error) {
+    console.error("Cache invalidation failed:", error);
+  }
+
+  return drink;
```
[the remark this answers is not in the document]
Comment (wKovacs64) [reply app/drinks/mutations.server.ts:32]: Slug uniqueness is validated before reaching this point, and SQLite insert failures on a single-admin app are extremely unlikely. The nested try/catch adds significant complexity for a near-zero probability event.

File: app/drinks/mutations.server.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  } else {
+    imageUrl = existingDrink.imageUrl;
+    imageFileId = existingDrink.imageFileId;
+  }
+
+  const drink = await updateDrinkRow(existingDrink.id, {
+    ...data,
+    imageUrl,
+    imageFileId,
+  });
```
[the remark this answers is not in the document]
Comment (wKovacs64) [reply app/drinks/mutations.server.ts:66]: The old image deletion is already in a try/catch (non-fatal), and if the DB update fails you'd have an orphaned new image either way. The suggested rewrite is ~30 lines of nested try/catch for a scenario that essentially never happens in a single-admin app.

File: playwright.config.ts [hunk 5]
```diff
@@ -25,5 +25,8 @@ export default defineConfig({
     timeout: 120000,
     stdout: "pipe",
     stderr: "pipe",
+    env: {
+      NODE_OPTIONS: "--import ./playwright/mocks.ts",
```
[the remark this answers is not in the document]
Comment (wKovacs64) [reply playwright.config.ts:29]: This project requires Node >=24 (per package.json engines). Node 24 has native TypeScript type stripping enabled by default, so `--import ./playwright/mocks.ts` works fine.

File: app/routes/admin.drinks.$slug.delete.tsx [hunk 6]
```diff
@@ -15,25 +13,18 @@ export async function action({ request, params }: Route.ActionArgs) {
...
-  } catch (error) {
-    // Cache invalidation failures shouldn't block the request
-    console.error("Cache invalidation failed:", error);
+    await deleteDrink(drink);
+  } catch {
+    session.flash("toast", {
+      kind: "error" as const,
+      message: `Failed to delete ${drink.title} — image cleanup failed`,
+    });
+    return data({ success: false }, { headers: { "Set-Cookie": await commitSession(session) } });
```
[the remark this answers is not in the document]
Comment (wKovacs64) [reply app/routes/admin.drinks.$slug.delete.tsx]: Good catch — generalized the error message and added status 500.

**Comment by wKovacs64** [comment]
## Non-test changes summary

1. **New `app/drinks/mutations.server.ts`** — extracted `createDrink`, `updateDrink`, `deleteDrink` orchestration (image upload/delete + DB write + cache purge) into a dedicated module
2. **`app/models/drink.server.ts`** — renamed `createDrink`/`updateDrink`/`deleteDrink` to `insertDrinkRow`/`updateDrinkRow`/`deleteDrinkRow` (now pure DB layer)
3. **Route handlers simplified:**
   - `admin.drinks.new.tsx` — collapsed ~25 lines of image upload + DB + cache into a single `createDrink()` call
   - `admin.drinks.$slug.edit.tsx` — same for updates; removed `existingImageUrl` hidden input logic; added stale image error toast
   - `admin.drinks.$slug.delete.tsx` — delegated to `deleteDrink()`; now redirects on success (was returning `{ success: true }`); shows error toast on failure instead of swallowing errors
4. **`app/admin/drink-form.tsx`** — removed the `existingImageUrl` hidden input (no longer needed)
5. **`app/search/minisearch.server.ts`** — `searchDrinks()` is now async, fetches its own data, and returns `Drink[]` instead of slugs
6. **`app/search/cache.server.ts`** — switched to relative import
7. **`app/routes/_app.search/route.tsx`** — simplified: calls `searchDrinks(q)` directly, removed manual slug→drink mapping
8. **`app/utils/imagekit.server.ts`** — removed `uploadImageOrPlaceholder` and test-mode guards; exported `uploadImage` directly
9. **`playwright/seed-data.ts`** — replaced `"test-placeholder"` fileIds with unique `"seed-fileId-*"` values
10. **`tsconfig.json`** — added `allowImportingTsExtensions: true`
